src/foundry/foundry.js/applications/basePlaceableHUD.d.ts

```diff
@@ -1,93 +1,89 @@
 /**
  * An abstract base class for displaying a heads-up-display interface bound to a Placeable Object on the canvas
- * @typeParam O - the type of the PlaceableObject
- * @typeParam P - the type of the options object
+ * @typeParam Object - the type of the PlaceableObject
+ * @typeParam Options - the type of the options object
  */
 declare abstract class BasePlaceableHUD<
-  O extends PlaceableObject<any>,
-  P extends Application.Options = Application.Options
-> extends Application<P> {
+  Object extends PlaceableObject<any> = PlaceableObject,
+  Options extends Application.Options = Application.Options
+> extends Application<Options> {
   /**
    * Reference a PlaceableObject this HUD is currently bound to
-   * @defaultValue `null`
+   * @defaultValue `undefined`
    */
-  object: O | null;
+  object: Object | undefined;
 
   /**
    * @override
    * @defaultValue
    * ```
-   * mergeObject(super.defaultOptions, {
+   * foundry.utils.mergeObject(super.defaultOptions, {
    *   classes: ["placeable-hud"],
    *   popOut: false
    * })
    * ```
    */
-  static get defaultOptions(): typeof Application['defaultOptions'];
+  static get defaultOptions(): Application.Options;
 
   /**
    * Convenience access for the canvas layer which this HUD modifies
    */
-  get layer(): O['layer'];
+  get layer(): Object['layer'] | undefined;
 
   /**
    * Bind the HUD to a new PlaceableObject and display it
    * @param object - A PlaceableObject instance to which the HUD should be bound
    */
-  bind(object: O): void;
+  bind(object: Object): void;
 
   /**
    * Clear the HUD by fading out it's active HTML and recording the new display state
    */
   clear(): void;
 
   /** @override */
-  _render(...args: Parameters<Application<P>['_render']>): Promise<void>;
+  _render(...args: Parameters<Application<Options>['_render']>): Promise<void>;
 
-  /**
-   * @override
-   * @defaultValue
-   * ```
-   * mergeObject(this.object.data, {
-   *   id: this.id,
-   *   classes: this.options.classes.join(' '),
-   *   appId: this.appId,
-   *   isGM: game.user.isGM,
-   *   icons: CONFIG.controlIcons
-   * });
-   * ```
-   */
-  getData(options?: Partial<P>): foundry.utils.Duplicated<O['data']> & {
-    id: Application['id'];
+  /** @override */
+  getData(options?: Partial<Options>): ReturnType<Object['data']['toJSON']> & {
+    id: BasePlaceableHUD['id'];
```

Comment:
I think we could just use the primitive types instead of referencing it. IMO that really only makes sense for complicated types.